tests/test_ubuntu_advantage.py

```diff
@@ -19,3 +27,74 @@ def test_get_args_disable(self):
         """The 'disable-esm' action can be called."""
         args = ubuntu_advantage.get_args(['disable-esm'])
         self.assertEqual('disable-esm', args.action)
+
+    def test_get_list_file(self):
+        """get_list_file returns the name of the sources.list.d file."""
+        self.assertEqual(
+            ubuntu_advantage.get_list_file(),
+            '/etc/apt/sources.list.d/ubuntu-esm-precise.list')
+
+    def test_write_list_file(self):
+        """write_list_file writes the sources.list.d file."""
+        list_file = self.tempdir.join('sample.list')
+        ubuntu_advantage.write_list_file(list_file, 'user:pass')
+        with open(list_file) as fh:
+            content = fh.read()
+        expected = (
+            'deb https://user:pass@extended.security.staging.ubuntu.com'
+            '/ubuntu precise main\n'
+            '# deb-src https://user:pass@extended.security.staging.ubuntu.com'
+            '/ubuntu precise main\n')
+        self.assertEqual(expected, content)
+
+    def test_enable_esm(self):
+        """enable_esm adds the repository to sources lists."""
+        ubuntu_advantage.enable_esm(
+            'user:pass', lists_dir=self.tempdir.path, print=self.print)
+        list_file = self.tempdir.join('ubuntu-esm-precise.list')
+        with open(list_file) as fh:
+            content = fh.read()
+        expected = (
+            'deb https://user:pass@extended.security.staging.ubuntu.com'
+            '/ubuntu precise main\n'
+            '# deb-src https://user:pass@extended.security.staging.ubuntu.com'
+            '/ubuntu precise main\n')
+        self.assertEqual(expected, content)
+        # a message is printed out
+        self.assertEqual(
+            ['Ubuntu ESM repository enabled.'
+             '  Run "sudo apt-get update" to update lists.'],
+            self.messages)
+
+    def test_disable_esm(self):
+        """disabl_esm renames the lists file as .save."""
+        ubuntu_advantage.enable_esm(
+            'user:pass', lists_dir=self.tempdir.path, print=self.print)
+        list_file = self.tempdir.join('ubuntu-esm-precise.list')
+        with open(list_file) as fh:
+            original_content = fh.read()
+
+        ubuntu_advantage.disable_esm(
+            lists_dir=self.tempdir.path, print=self.print)
+        # the file has moved
+        self.assertFalse(os.path.exists(list_file))
+        # the original content is still there
+        with open(list_file + '.save') as fh:
+            content = fh.read()
+        self.assertEqual(original_content, content)
+        # a message is printed out
+        self.assertIn(
+            'Ubuntu ESM repository enabled.'
```

Comment:
Oops, I missed this the first time. This should be looking for 'Ubuntu ESM repository disabled.'. This test passes because it calls enable too so both messages are in self.message. My quick fix for this is https://pastebin.canonical.com/185715/.